Src/Plugins/Security/Security.SharedStrings.txt

```diff
@@ -38,6 +38,6 @@
   $SEC101/046.SqlConnectionStringAdo=(?i)(((server|data source)\s*=\s*[^;"<]{0,50}|(initial catalog|database)\s*=\s*[^;"<]{0,50}|(user id|uid)\s*=\s*[^;"<]{0,50}|(password|pwd)\s*=\s*[^;"<]{0,50}).+?){4}
   $SEC101/046.SqlConnectionStringJdbc=(?i)jdbc:sqlserver:\/\/(?<host>[^;"<]+);(((database\s*=\s*(?<database>[^;"<]{1,128}))|(user\s*=\s*(?<account>[^@]+)@[^;"<]+)|(password\s*=\s*(?<password>[^;"<]+))).{1,50}?){3}
   $SEC101/046.SqlConnectionStringPhp=(?i)sqlsrv:server\s*=\s*(?<host>[^;"<]+);\s*Database\s*=\s*(?<database>[^\"]+)",\s*"(?<account>[^"]+)",\s*"(?<password>[^"]+)"
-  $SEC101/047.MySqlConnectionStringAdo=(?i)(Port\s*=\s*([0-9]{4,5}).*)?(((Server\s*=\s*(?<host>[\w\-.]{3,90}))|(Uid=(?-i)(?<account>[a-z\@\-]{1,120})(?i))|(Pwd=(?<password>[^;]{8,128}))).*?){3}(.*Port\s*=\s*([0-9]{4,5}))?
+  $SEC101/047.MySqlConnectionStringAdo=(?i)(Port\s*=\s*([0-9]{4,5}).*)?(((Server\s*=\s*(?<host>[\w\-.]{3,90}))|(Uid=(?-i)(?<account>[a-z\@\-]{1,120})(?i))|(Pwd=(?<password>[^;"]{8,128}))).*?){3}(.*Port\s*=\s*([0-9]{4,5}))?
```

Comment:
>Pwd [](start = 156, length = 3)

This change breaks us for patterns where the password contains a double quote, or when the double quote is escaped. Right?

Reply:
right. do u know a better way to do this? This would happen for all the sql strings (MySQL, Postgres and MSSQL)

---
In reply to: [576398780](https://github.com/microsoft/sarif-pattern-matcher/pull/214#discussion_r576398780) [](ancestors = 576398780)